File: src/services/autocomplete/__tests__/autocomplete_sigil_tests.ml
(*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *)

open OUnit2

let tests =
  [
    "add"
    >::: [
           ( "empty_line" >:: fun ctxt ->
             let contents = "// @flow\n\nfoo\n\nbar" in
             let expected = "// @flow\n\nfoo\nAUTO332\nbar" in
             let (actual, broader_context) = Autocomplete_sigil.add contents 4 0 in
             assert_equal ~ctxt ~printer:(fun x -> x) expected actual;
             let expected = "foo\nAUTO332\nbar" in
             assert_equal ~ctxt ~printer:(fun x -> x) expected broader_context
           );
           ( "last_line" >:: fun ctxt ->
             let contents = "// @flow\n" in
             let expected = "// @flow\nAUTO332" in
             let (actual, broader_context) = Autocomplete_sigil.add contents 2 0 in
             assert_equal ~ctxt ~printer:(fun x -> x) expected actual;
             let expected = "// @flow\nAUTO332" in
             assert_equal ~ctxt ~printer:(fun x -> x) expected broader_context
           );
         ];
  ]

File: rust_port/crates/flow_services_autocomplete/src/autocomplete_sigil_tests.rs
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

//! Port of `services/autocomplete/__tests__/autocomplete_sigil_tests.ml`

#[cfg(test)]
mod tests {
    use crate::autocomplete_sigil;

    #[test]
    fn empty_line() {
        let contents = "// @flow\n\nfoo\n\nbar";
        let expected = "// @flow\n\nfoo\nAUTO332\nbar";
        let (actual, broader_context, _) = autocomplete_sigil::add(None, contents, 4, 0);
        assert_eq!(expected, actual);
        let expected = "foo\nAUTO332\nbar";
        assert_eq!(expected, broader_context);
    }

    #[test]
    fn last_line() {
        let contents = "// @flow\n";
        let expected = "// @flow\nAUTO332";
        let (actual, broader_context, _) = autocomplete_sigil::add(None, contents, 2, 0);
        assert_eq!(expected, actual);
        let expected = "// @flow\nAUTO332";
        assert_eq!(expected, broader_context);
    }
}
